### PR TITLE
Update bindings for 0.24

### PR DESCRIPTION
**Pull request by amaanq**
Namely:
- CMake support
- restructure some bindings
- migrate to tree-sitter.json

### CODE REVIEW

File: tree-sitter.json [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        "queries/injections-text.scm"
+      ],
+      "tags": "queries/tags.scm"
+    },
+    {
+      "name": "php_only",
+      "camelcase": "PHP_Only",
+      "scope": "source.php",
+      "path": "php_only",
+      "external-files": "common/scanner.h"
```
Comment (calebdw) [inline tree-sitter.json:24]: @amaanq,

This should have queries too right?

```suggestion
      "external-files": "common/scanner.h",
            "file-types": [
        "php"
      ],
      "highlights": "queries/highlights.scm",
      "injections": "queries/injections.scm",
      "tags": "queries/tags.scm"
```

Comment (amaanq) [reply tree-sitter.json:24]: yes, good catch

**Comment by amaanq** [comment]
I'll leave it to you to fix the failing file @calebdw :)

**Comment by calebdw** [comment]
> I'll leave it to you to fix the failing file

Issue is that unary op expressions (-1) aren't supported in enums yet, I'll send a fix.

By the way, would it be possible to update the parse action to show any failed files up front? It's kind of a pain to have to scroll through miles of lines to see which one failed

https://github.com/tree-sitter/tree-sitter-php/actions/runs/11339991148/job/31535650336

**Comment by amaanq** [comment]
> > I'll leave it to you to fix the failing file
> 
> Issue is that unary op expressions (-1) aren't supported in enums yet, I'll send a fix.
> 
> By the way, would it be possible to update the parse action to show any failed files up front? It's kind of a pain to have to scroll through miles of lines to see which one failed
> 
> https://github.com/tree-sitter/tree-sitter-php/actions/runs/11339991148/job/31535650336

Click summary on the top left

**Comment by amaanq** [comment]
> > I'll leave it to you to fix the failing file
> 
> Issue is that unary op expressions (-1) aren't supported in enums yet, I'll send a fix.
> 
> By the way, would it be possible to update the parse action to show any failed files up front? It's kind of a pain to have to scroll through miles of lines to see which one failed
> 
> https://github.com/tree-sitter/tree-sitter-php/actions/runs/11339991148/job/31535650336

Click summary on the top left

**Comment by calebdw** [comment]
Do we no longer run the local tests in ci? It looks like we just parse example files

**Comment by amaanq** [comment]
> Do we no longer run the local tests in ci? It looks like we just parse example files
> 
> (Sorry for deleting your comment---it came through twice on my end so I tried to clean up the duplicate and now it looks like none survived 😬)

Hm what do you mean by the local tests? iirc `tree-sitter test` is still invoked, as well as tests for every binding.

And no worries, I was on mobile with poor connection so it sent twice

**Comment by calebdw** [comment]
Unless I'm missing something I don't see `tree-sitter test` as part of these workflows:

image

**Comment by amaanq** [comment]
It's a step in the workflow 
image

**Comment by calebdw** [comment]
Ah, I see it now, it was nested in between the clones and I skipped over it thinking it was the example parsing:

```yaml
      - name: Run tests
        uses: tree-sitter/parser-test-action@v2
        with:
          test-rust: true
          test-node: true
          test-python: true
          test-go: true
          test-swift: true
```

Everything looks good to me, thanks for all your hard work on tree-sitter! I love the more frequent releases :)

**Comment by calebdw** [comment]
Do you know why node v23 isn't downloading?

https://github.com/tree-sitter/tree-sitter-php/actions/runs/11348329979/job/31562460626

**Pull request merged**